server/graphql/v2/query/collection/OrdersCollectionQuery.ts

```diff
@@ -37,165 +38,197 @@ const getJoinCondition = (
-    },
-    maxAmount: {
-      type: GraphQLInt,
-      description: 'Only return orders where the amount is lower than or equal to this value (in cents)',
-    },
-    dateFrom: {
-      type: GraphQLDateTime,
-      description: 'Only return orders that were created after this date',
-    },
-    dateTo: {
-      type: GraphQLDateTime,
-      description: 'Only return orders that were created after this date',
-    },
-    searchTerm: {
-      type: GraphQLString,
-      description: 'The term to search',
-    },
+export const OrdersCollectionArgs = {
+  ...CollectionArgs,
+  includeHostedAccounts: {
+    type: GraphQLBoolean,
+    description: 'If account is a host, also include hosted accounts orders',
   },
-  async resolve(_: void, args, req: express.Request): Promise<CollectionReturnType> {
-    const where = { [Op.and]: [] };
-    const include: Includeable[] = [
-      { association: 'fromCollective', required: true, attributes: [] },
-      { association: 'collective', required: true, attributes: [] },
-    ];
-
-    // Check arguments
-    if (args.limit > 1000 && !req.remoteUser?.isRoot()) {
-      throw new Error('Cannot fetch more than 1,000 orders at the same time, please adjust the limit');
-    }
-
-    // Load accounts
-    if (args.account) {
-      const fetchAccountParams = { loaders: req.loaders, throwIfMissing: true };
-      const account = await fetchAccountWithReference(args.account, fetchAccountParams);
+  includeIncognito: {
+    type: GraphQLBoolean,
+    description: 'Whether to include incognito orders. Must be admin or root. Only with filter null or OUTGOING.',
+    defaultValue: false,
+  },
+  filter: {
+    type: AccountOrdersFilter,
+    description: 'Account orders filter (INCOMING or OUTGOING)',
+  },
+  frequency: {
+    type: ContributionFrequency,
+    description: 'Use this field to filter orders on their frequency (ONETIME, MONTHLY or YEARLY)',
+  },
+  status: {
+    type: OrderStatus,
+    description: 'Use this field to filter orders on their statuses',
+  },
+  orderBy: {
+    type: new GraphQLNonNull(ChronologicalOrderInput),
+    description: 'The order of results',
+    defaultValue: CHRONOLOGICAL_ORDER_INPUT_DEFAULT_VALUE,
+  },
+  minAmount: {
+    type: GraphQLInt,
+    description: 'Only return orders where the amount is greater than or equal to this value (in cents)',
+  },
+  maxAmount: {
+    type: GraphQLInt,
+    description: 'Only return orders where the amount is lower than or equal to this value (in cents)',
+  },
+  dateFrom: {
+    type: GraphQLDateTime,
+    description: 'Only return orders that were created after this date',
+  },
+  dateTo: {
+    type: GraphQLDateTime,
+    description: 'Only return orders that were created after this date',
+  },
+  searchTerm: {
+    type: GraphQLString,
+    description: 'The term to search',
+  },
+  tierSlug: { type: GraphQLString },
```

Comment:
Should we have this one deprecated and accept a tier reference instead?

Reply:
Good idea. I can follow up.